LiveStream/Service/LiveStreamService.swift

```diff
@@ -54,6 +54,50 @@ public struct LiveStreamService: LiveStreamServiceProtocol {
       })
     }
   }
+
+  public func fetchEvents(forProjectId projectId: Int, uid: Int?) ->
+    SignalProducer<LiveStreamEventsEnvelope, LiveApiError> {
+
+    return SignalProducer { (observer, disposable) in
+      let uidString = uid
+        .flatMap { "?uid=\($0)" }
+        .coalesceWith("")
+
+      let urlString = "\(Secrets.LiveStreams.Api.base)/projects/\(projectId)\(uidString)"
```

Comment:
this we cant really test this function i'd like to do something a little safer around url construction. we can use `URLComponents` for the query params 

```swift
let url = URL(string: Secrets.LiveStreams.Api.base)?.appendingPathComponent("projects/\(projectId)")
var components = url.flatMap { URLComponents(url: $0, resolvingAgainstBaseURL: false) }
components?.queryItems = uid.map { [URLQueryItem(name: "uid", value: "\($0)")] }

guard let url = components?.url ...
```

did that without the type checker, but something along those lines...